src/detect-engine-loader.h

```diff
@@ -43,10 +43,14 @@ typedef struct DetectLoaderTask_ {
 
 typedef struct DetectLoaderControl_ {
     int id;
-    int result;     /* 0 for ok, error otherwise */
-    ThreadVars *tv; /* loader threads threadvars - for waking them up */
-    SCMutex m;
-    TAILQ_HEAD(, DetectLoaderTask_) task_list;
+    ThreadVars *tv; /**< loader threads threadvars - for waking them up */
+
+    /** struct to group members and mutex */
+    struct {
+        SCMutex m;  /**< mutex protects result and task_list */
+        int result; /**< 0 for ok, error otherwise */
+        TAILQ_HEAD(, DetectLoaderTask_) task_list;
+    };
```

Comment:
This is more legible. Thank you.
Although, this creates some holes in this struct compared to last PR.

```
struct DetectLoaderControl_ {
	int                        id;                   /*     0     4 */

	/* XXX 4 bytes hole, try to pack */

	ThreadVars *               tv;                   /*     8     8 */
	struct {
		pthread_mutex_t    m;                    /*    16    40 */
		int                result;               /*    56     4 */

		/* XXX 4 bytes hole, try to pack */

		/* --- cacheline 1 boundary (64 bytes) --- */
		struct {
			struct DetectLoaderTask_ * tqh_first; /*    64     8 */
			struct DetectLoaderTask_ * * tqh_last; /*    72     8 */
		} task_list;                             /*    64    16 */
	};                                               /*    16    64 */

	/* size: 80, cachelines: 2, members: 3 */
	/* sum members: 76, holes: 1, sum holes: 4 */
	/* last cacheline: 16 bytes */
};
```

Is this ok?

Reply:
yes, it is not perf critical or memory critical, so code organization/readability can be leading here